core/storage/base_model/gae_models.py

```diff
@@ -538,10 +540,14 @@ def get_multi_versions(cls, entity_id, version_numbers):
             version_numbers: list(int). List of version numbers.
 
         Returns:
-            list(VersionedModel|None). Model instances representing the given
+            list(VersionedModel). Model instances representing the given
                 versions.
         """
         instances = []
+        try:
+            cls.get(entity_id)
+        except Exception as  e:
+            logging.error(e)
```

Comment:
Why log? If this is not expected to happen, raise directly.

In designing this particular API, you'll need to figure out: (a) is the client expected to ensure that the exp id and version numbers are valid, or (b) is this method supposed to take care of that? If (a), specify that constraint clearly in the docstring. If (b), specify what is returned/raised in cases of error, and add tests.

Reply:
Done.